### DISCUSSION TITLE
Strange interaction between Shadow and FluentWait

### DISCUSSION THREAD

**Discussion by zamioku** [sukgu/shadow-automation-selenium#43] [Q&A] [answered]
Question: Can someone show me an example of waiting for a shadow element to be visible?

I've tried using Selenium's FluentWait but, if I use the Shadow.setExplicitWait() it seems to take into consideration this wait time as polling time, and ignores the timeout set on my fluent wait.

Here's a code snippet:
```
Wait<WebDriver> myWait = new FluentWait()<>(driver)
    .withTimeout(Duration.ofSeconds(15))
    .pollingEvery(Duration.ofSeconds(1))
    .ignoring(NoSuchElementException.class);

myWait.until(ExpectedConditions.visibilityOf(shadow.findElement("#id")));
```

**Comment by sukgu** [comment, marked as answer]
For shadow objects Selenium wait won't work. For Shadow objects you can use `void setExplicitWait(int seconds, int pollingTime) throws Exception`. This has wait time in seconds as 1st arg and pollingTime as 2nd args.

To use this you need to set first explicit wait using `setExplicitWait` method and then try finding the element using `shadow.findElement` method


```
Shadow shadow = new Shadow(driver);
shadow.setExplicitWait(10, 2);
WebElement element = shadow.findElement("div@user");
//now you can reset the wait to 0
shadow.setExplicitWait(0, 0);
```

**Reply by zamioku** [reply]
Thank you for your reply. This helped me a lot!
